state/protocol/inmem/snapshot.go

```diff
@@ -0,0 +1,72 @@
+package inmem
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/state/protocol/seed"
+)
+
+var (
+	_ protocol.Snapshot   = new(Snapshot)
+	_ protocol.EpochQuery = new(Epochs)
+	_ protocol.Epoch      = new(Epoch)
+	_ protocol.Cluster    = new(Cluster)
+)
+
+// Snapshot is a memory-backed implementation of protocol.Snapshot. The snapshot
+// data is stored in the embedded encodable snapshot model, which defines the
+// canonical structure of an encoded snapshot for the purposes of serialization.
+type Snapshot struct {
+	enc EncodableSnapshot
+}
+
+func (s *Snapshot) Head() (*flow.Header, error) {
```

Comment:
I think these methods are not required to be pointer receiver. They do not change affect `s` and safer to be by value.